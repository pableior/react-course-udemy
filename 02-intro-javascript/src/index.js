
const firstName = 'Pablo';
const lastName = 'Gonzalez';
let diceValue = 5;

diceValue = 6;
console.log(firstName, lastName, diceValue);

if (true) {
    const firstName = 'Peter';
    let diceValue = 1;
    console.log(firstName, diceValue);
}

console.log(firstName, diceValue);

