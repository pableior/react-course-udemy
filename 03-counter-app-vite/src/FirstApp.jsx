import PropTypes from 'prop-types';
// yarn add prop-types

const sayHi = (name) => {
    return `Hola ${name}, buen dia!`;
}

const newMsg = {
    a: 'hola',
    b: 'Pablo',
}

export const FirstApp = ({title, subTitle, name}) => {

    return (
        <>
            {/* <h1> {sayHi('Pablo')} </h1> */}
            {/* <h1>{JSON.stringify(newMsg)}</h1> */}
            <h1>{title}</h1>
            <p> {subTitle} </p>
            <p> {name} </p>
        </>
    );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Pablo Gonzalez',
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo', //va a dar error porque el tipo es del tipo number
}

