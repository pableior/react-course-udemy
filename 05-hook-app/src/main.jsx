import React from 'react'
import ReactDOM from 'react-dom/client'
import { BrowserRouter } from "react-router-dom";
import './index.css'
import { MainApp } from './09-useContext/MainApp'
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
//import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScree } from './04-useRef/FocusScree'
// import { Layout } from './05-useLayoutEffect/Layout'
// import { Memorize } from './06-memos/Memorize'
// import { MemoHook } from './06-memos/MemoHook'
// import { CallbackHooks } from './06-memos/CallbackHooks'
// import { Padre } from './07-tarea/Padre'

//import './08-useReducer/intro-reducer';
//import { TodoApp } from './08-useReducer/TodoApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
  {/*<React.StrictMode>*/}
    <MainApp />
  {/*</React.StrictMode>,*/}
  </BrowserRouter>
)
