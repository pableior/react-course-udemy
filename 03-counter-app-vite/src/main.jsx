import React from 'react';

import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import './styles.css';
import { CounterApp } from './CounterApp';
ReactDOM.createRoot(document.getElementById('root')).render (
    <React.StrictMode>
        <FirstApp value = {15} />
    </React.StrictMode>
);