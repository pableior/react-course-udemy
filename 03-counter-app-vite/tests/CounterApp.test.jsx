import { render, screen } from '@testing-library/react';
import { CounterApp } from "../src/CounterApp";

describe('test on component counter app', () => { 
    
    test('must match to snapshot', () => {
        const value = 100;
        const {container} = render(<CounterApp value={value}/>);
        expect(screen).toMatchSnapshot();
        expect(container).toMatchSnapshot();
    });

    test('must contain initial value text', () => {
        const value = 100;
        const {container} = render(<CounterApp value={value}/>);
        expect(screen.getAllByText(`${value}`).length).toBe(1);
        expect(screen.getByText(value)).toBeTruthy();
        expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain('100');


    });
});