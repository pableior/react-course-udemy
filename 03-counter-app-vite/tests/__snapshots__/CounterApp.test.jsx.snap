// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on component counter app must match to snapshot 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`test on component counter app must match to snapshot 2`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
